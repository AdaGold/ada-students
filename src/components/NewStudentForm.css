label { 
  margin-right: 10px;
  margin-left: auto;
}

form div {
  display: grid;
  grid-template-columns: 100px 1fr;
  width: 400px;
  margin: 5px;
}