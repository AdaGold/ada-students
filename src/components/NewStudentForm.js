import React, { useState } from 'react';
import './NewStudentForm.css';


const NewStudentForm = (props) => {
  const [formFields, setFormFields] = useState({
    fullName: '',
    email: '',
  });

  // event handlers
  const onInputChange = (event) => {
    console.log(`Changing field ${ event.target.name } to ${ event.target.value }`);
    const newFormFields = {
      ...formFields,
    }
    newFormFields[event.target.name] = event.target.value;
    setFormFields(newFormFields);
  }

  const onFormSubmit = (event) => {
    event.preventDefault();

    props.addStudentCallback(formFields);

    setFormFields({
      fullName: '',
      email: '',
    });
  };

  // validate email
  const emailValid = () => {
    return formFields.email.match(/\S+@\S+/) || formFields.email === '';
  }

  return (
    <form
      className="new-student-form"
      onSubmit={onFormSubmit}
      data-testid="new-student-form"
    >
      <div>
        <label htmlFor="fullName">Name:</label>
        <input
          id="fullName"
          name="fullName"
          onChange={onInputChange}
          value={formFields.fullName}
        />
      </div>
      <div>
        <label htmlFor="email">Email:</label>
        <input
          id="email"
          name="email"
          onChange={onInputChange}
          value={formFields.email}
          className={emailValid() ? "valid" : "invalid"}
        />
      </div>
      <input
        type="submit"
        value="Add Student"
      />
    </form>
  );
}

export default NewStudentForm;