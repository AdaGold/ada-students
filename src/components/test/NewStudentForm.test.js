// src/components/test/NewStudentForm.test.js
import React from 'react';
import { render, cleanup } from '@testing-library/react'
import NewStudentForm from '../NewStudentForm';

describe('NewStudentForm', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <NewStudentForm
        addStudentCallback={() => { }}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
});