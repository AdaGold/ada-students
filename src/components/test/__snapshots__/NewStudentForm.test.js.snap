// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewStudentForm that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <form
    class="new-student-form"
  >
    <div>
      <label
        for="fullName"
      >
        Name:
      </label>
      <input
        name="fullName"
        value=""
      />
    </div>
    <div>
      <label
        for="email"
      >
        Email:
      </label>
      <input
        class="valid"
        name="email"
        value=""
      />
    </div>
    <input
      type="submit"
      value="Add Student"
    />
  </form>
</DocumentFragment>
`;
